com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ModelBuilder.java

```diff
@@ -115,18 +120,59 @@ else if(!modelDescBuilder.getAnimations().isEmpty()) {
         Model.Builder model = Model.newBuilder();
         model.setRigScene(task.output(1).getPath().replace(this.project.getBuildDirectory(), ""));
 
-        BuilderUtil.checkResource(this.project, resource, "material", modelDescBuilder.getMaterial());
-        model.setMaterial(BuilderUtil.replaceExt(modelDescBuilder.getMaterial(), ".material", ".materialc"));
+        if (modelDescBuilder.getMaterialsCount() > 0)
+        {
+            for (Material material : modelDescBuilder.getMaterialsList()) {
+                Material.Builder materialBuilder = Material.newBuilder();
 
-        List<String> newTextureList = new ArrayList<String>();
-        for (String t : modelDescBuilder.getTexturesList()) {
-            if (t.isEmpty())
-                continue; // TODO: Perhaps we can check if the material expects textures?
+                BuilderUtil.checkResource(this.project, resource, "material", material.getMaterial());
+                materialBuilder.setName(material.getName());
+                materialBuilder.setMaterial(BuilderUtil.replaceExt(material.getMaterial(), ".material", ".materialc"));
+
+                List<Texture> texturesList = new ArrayList<>();
+                for (Texture t : material.getTexturesList()) {
+                    BuilderUtil.checkResource(this.project, resource, "texture", t.getTexture());
+
+                    Texture.Builder textureBuilder = Texture.newBuilder(t); // create from existing message
+                    textureBuilder.setTexture(ProtoBuilders.replaceTextureName(t.getTexture()));
+                    texturesList.add(textureBuilder.build());
+                }
+
+                materialBuilder.addAllTextures(texturesList);
+                model.addMaterials(materialBuilder);
+            }
+        } else {
+            // Deprecated workflow
+            String singleMaterial = modelDescBuilder.getMaterial();
+            if (!singleMaterial.isEmpty()) {
+
+                logger.log(Level.WARNING, String.format("Model %s uses deprecated material format. Please resave in the editor!", task.input(0).getAbsPath()));
```

Comment:
Should we silence this warning until the editor actually migrates the file formats?